.github/workflows/ci.yml

```diff
@@ -194,3 +194,40 @@ jobs:
         uses: codecov/codecov-action@v3
         with:
           files: lcov.info
+
+  std-tests:
+    env:
+      NUSHELL_CARGO_TARGET: ci
+
+    strategy:
+      fail-fast: true
+      matrix:
+        platform: [windows-latest, macos-latest, ubuntu-20.04]
+        style: [default, dataframe]
+        rust:
+          - stable
+        include:
+          - style: default
+            flags: ""
+          - style: dataframe
+            flags: "--features=dataframe"
+        exclude:
+          # only test dataframes on Ubuntu (the fastest platform)
+          - platform: windows-latest
+            style: dataframe
+          - platform: macos-latest
+            style: dataframe
+
+    runs-on: ${{ matrix.platform }}
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Rust toolchain and cache
+        uses: actions-rust-lang/setup-rust-toolchain@v1.4.3
+
+      - name: Install nushell
+        run: cargo install --path . ${{ matrix.flags }}
```

Comment:
Could we reuse the virtualenv runner that already has this setup to save some time?

Reply:
oh that would be super nice :+1: 

i can have a look after lunch.

Reply:
ooh do you mean getting rid of the `std-tests` and simply put the `nu .../tests.nu` inside `python-virtualenv`?

Reply:
> ooh do you mean getting rid of the `std-tests` and simply put the `nu .../tests.nu` inside `python-virtualenv`?

Yeah we could combine them. It is a bit worse from a reporting perspective but we could make the std tests the first steps and folllow up with the virtualenv stuff.

Reply:
maybe we could reuse the artifact, i.e. the `nu` binary executable, generated by
```yaml
      - name: Install Nushell
        run: cargo install --locked --path=. --profile ci --no-default-features
```
in `jobs.python-virtualenv` ? :thinking:

Reply:
shoud i move `nu tests.nu` inside `python-virtualenv` or the other way around? :yum:

Reply:
I would recommend changing the name to something that reflects both. I think you can reuse the `--profile ci` build, which is debug without internal symbols which saves some time and cache.

Reply:
a name that reflects both, gotcha :+1:

Reply:
mm that did not appear to trigger the CI :thinking: